### ISSUE TITLE
Remove battery node updates from globalstep technic_run

### ISSUE THREAD

**Issue by S-S-X** [mt-mods/technic#108]
Remove battery node updates from technic_run and use nodetimers to update batteries.
For actual charge status create single large virtual battery for network so that technic_run can base calculations on single value.

All of following is found from single function, refactor all these loops. Should be able to reduce loops significantly and also even out load to nodetimers:
https://github.com/mt-mods/technic/blob/f0924dfc7af187ec61ec46d31f7465ebd8675340/technic/machines/network.lua#L423-L445
https://github.com/mt-mods/technic/blob/f0924dfc7af187ec61ec46d31f7465ebd8675340/technic/machines/network.lua#L464-L478
https://github.com/mt-mods/technic/blob/f0924dfc7af187ec61ec46d31f7465ebd8675340/technic/machines/network.lua#L515-L524
https://github.com/mt-mods/technic/blob/f0924dfc7af187ec61ec46d31f7465ebd8675340/technic/machines/network.lua#L541-L553
https://github.com/mt-mods/technic/blob/f0924dfc7af187ec61ec46d31f7465ebd8675340/technic/machines/network.lua#L563-L572

Do this after #96 is merged.

**Comment by S-S-X** [comment]
Battery box node swapping (power meter texture) uses nodetimer so it should only happen in active areas, all other functionality still happens within globalstep loop.
Start at least, for some weir reason it did seem that I managed to cut over 40% from technic_run loop but changes I made there should not really give that large boost so not sure why I actually got timings like that (probably some error as I've been running some very basic profiling in background all the time).
But interesting that something happened, worst case might be that Icause it to skip some important tasks and got "boost" from that :smile: 

Formspec was updated too, see #112

**Comment by S-S-X** [comment]
It would be possible to remove all metadata from batteries and only store values for charge and max_charge at network level.
However that would make it impossible to detach battery from network by digging cable that connects it to network and also destroying network data would either reset batteries or would require complex handling for batteries, I think having metadata for that is good so maybe not remove that (`internal_EU_charge`).

### Next changes for battery handling
Batteries are not functional when not attached to network which means that `*_EU_supply` and `*_EU_demand` meta data values for batteries could be removed completely. Possibly also `internal_EU_charge_max`. Those are still there but not used that much after last PR #126 

This requires always balancing charge level between batteries attached to network, it is done already to some degree but is bit buggy. Take battery upgrades into account as those allow having different maximum capacity for batteries, therefore it would be best to allow battery to set actual charge values and only inform batteries about network charge level as percentage value. That would also fix current glitches with having 1000kEU max charge battery with 1200kEU actual charge.

**Comment by S-S-X** [comment]
Added new issue for rest of it, a lot of unnecessary updates were already removed from globalstep and some of the remaining stuff is about redesigning some of the energy distribution systems.
